### PR TITLE
Pytest skip 2

### PR DESCRIPTION
**Pull request by KuuCi**
# What does this PR do?

This PR is meant to fix flakiness in composer daily test. In particular, it's suspected that pytest skip, test tmp directories are being cleared before being used during clean up, leading to directory not found. All this change does is adds a time.sleep(10) before skipping to make sure all thread directories are created before clean up process starts

# What issue(s) does this change relate to?

N/A

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Comment by KuuCi** [comment]
https://github.com/mosaicml/composer/actions/runs/9749564754
https://github.com/mosaicml/composer/actions/runs/9749962833

**Comment by KuuCi** [comment]
(saw it was still flaky with 5 seconds :( )

File: tests/trainer/test_fsdp_checkpoint.py [hunk 1]
```diff
@@ -836,6 +836,8 @@ def test_fsdp_partitioned_state_dict_load(
     if weights_only and autoresume:
         pytest.skip('Weights only with autoresume is not supported')
     if (use_tp or use_hsdp) and version.parse(torch.__version__) < version.parse('2.3.0'):
+        import time
+        time.sleep(10)
```
Comment (mvpatel2000) [inline tests/trainer/test_fsdp_checkpoint.py:840]: can u explain why?

**Comment by KuuCi** [comment]
https://github.com/mosaicml/composer/actions/runs/9751225339
https://github.com/mosaicml/composer/actions/runs/9751681384

File: tests/trainer/test_fsdp_checkpoint.py [hunk 2]
```diff
@@ -836,6 +836,7 @@ def test_fsdp_partitioned_state_dict_load(
     if weights_only and autoresume:
         pytest.skip('Weights only with autoresume is not supported')
     if (use_tp or use_hsdp) and version.parse(torch.__version__) < version.parse('2.3.0'):
+        dist.barrier()
```
Comment (mvpatel2000) [inline tests/trainer/test_fsdp_checkpoint.py:839]:
```suggestion
        dist.barrier()  # Sync to avoid race conditions on cleaning up tmp_path
```

**Review by mvpatel2000 — APPROVED** [review]

**Pull request merged**